### PR TITLE
Adjust anvil combining for tools

### PR DESCRIPTION
**Pull request by X1nto**
In vanilla Minecraft, when combining tools/armor with the same enchants, the level of the enchantment in the combined item is produced with this formula:
  - If the levels are the same on both the first and the second item, the combined item will have the enchantment level of [the level that both items have] + 1. i.e: if we have 2 iron chestplates with protection 2 on both of them, combining those will give us an iron chestplate with protection 3.
  - If levels are different, the combined item will have the highest level of the 2 provided items. i.e: if we have 2 diamond boots with protection 1 and 2, the combined boots will have protection 2.

This PR ensures the same happens in Minetinker when combining tools with the same enchantments.

### CODE REVIEW

File: src/main/java/de/flo56958/minetinker/listeners/AnvilListener.java [hunk 1]
```diff
@@ -179,10 +180,28 @@ public void onAnvilPrepare(@NotNull final PrepareAnvilEvent event) {
...
+				final List<Modifier> newToolMods = modManager.getToolMods(newTool);
+				final List<Modifier> item2Mods = modManager.getToolMods(item2);
+				final List<Modifier> sharedModifiers = item2Mods.stream().filter(newToolMods::contains).toList();
+				final List<Modifier> uniqueModifiers = item2Mods.stream().filter(Predicate.not(newToolMods::contains)).toList();
+
+				for (var shared : sharedModifiers) {
+					final int newToolModLevel = modManager.getModLevel(newTool, shared);
+					final int item2ModLevel = modManager.getModLevel(item2, shared);
+
+					if (item2ModLevel > newToolModLevel) { //If the 2nd tool's modifier has a higher level, clamp the level to the 2nd tool's level.
```
Comment (Flo56958) [inline src/main/java/de/flo56958/minetinker/listeners/AnvilListener.java:192]: Is it necessary to do it the other way around as well?

Comment (Flo56958) [reply src/main/java/de/flo56958/minetinker/listeners/AnvilListener.java:192]: On further thinking, it is not necessary as no changes need to be made to the modifier on the tool, but maybe an explaining comment would help as it is not in plain site.

**Review by Flo56958 — COMMENTED** [review]
Looks great and would change the merging to something less exploitable.

One questions: How are the free slots and exp handled? Discarded on the second tool?

File: src/main/java/de/flo56958/minetinker/listeners/AnvilListener.java (hunk 1 above)
Comment (Flo56958) [reply src/main/java/de/flo56958/minetinker/listeners/AnvilListener.java:192]: @X1nto

Comment (X1nto) [reply src/main/java/de/flo56958/minetinker/listeners/AnvilListener.java:192]: Basically, if the item2's enchant level is higher than the item1's enchant level, the outcome level is the one from item2. if the levels are the same, the outcome is the item2 level + 1, and if item2's level is lower than item1's, the outcome is item1's level.

**Comment by X1nto** [comment]
> Looks great and would change the merging to something less exploitable.
> 
> One questions: How are the free slots and exp handled? Discarded on the second tool?

Free slots are ignored while XP is merged, just like how it was previously. Although, it would be possible for free slot ignoration to only work for shared modifiers, or vice versa

**Comment by X1nto** [comment]
By "ignored", I mean that the combination does not care whether the tools have free slots or not, as in the end, combining does not affect free slot count at all (except if XP merge causes a level up, I guess).

File: src/main/java/de/flo56958/minetinker/listeners/AnvilListener.java (hunk 1 above)
Comment (X1nto) [reply src/main/java/de/flo56958/minetinker/listeners/AnvilListener.java:193]: By default, the outcome is item1's level with checks for the remaining 2 situations added later, so the check for when item2 < item1 is not necessary

**Pull request merged**